src/test/scala/io/citrine/lolo/bags/CorrelationStudy.scala

```diff
@@ -0,0 +1,441 @@
+      observational = observational,
+      samplingNoise = samplingNoise,
+      rhoTrain = rhoTrain,
+      quadraticCorrelationFuzz = quadraticCorrelationFuzz,
+      rng = rng
+    )
+
+    saveChart(chart, fname)
+    saveRawData(
+      chart = chart,
+      fname = fname,
+      metric = metric,
+      variedParameter = variedParameter,
+      parameterValues = parameterValues,
+      testProblem = testProblem,
+      function = function,
+      numTrials = numTrials,
+      numTrain = numTrain,
+      numTest = numTest,
+      observational = observational,
+      samplingNoise = samplingNoise,
+      rhoTrain = rhoTrain,
+      quadraticCorrelationFuzz = quadraticCorrelationFuzz
+    )
+  }
+
+  def makeChart(
+                 variedParameter: VariedParameter,
+                 metric: Metric,
+                 parameterValues: Seq[Double],
+                 testProblem: TestProblem,
+                 function: TrueFunction,
+                 numTrials: Int,
+                 numTrain: Int,
+                 numTest: Int,
+                 observational: Boolean,
+                 samplingNoise: Double,
+                 rhoTrain: Double,
+                 quadraticCorrelationFuzz: Double,
+                 rng: Random
+               ): XYChart = {
+    // the linear problem compares label 0 to label 1, and the quadratic problem compares label 0 to label 2
+    val index: Int = testProblem match {
+      case Linear => 1
+      case Quadratic => 2
+    }
+    // Select the figures of merit based on the desired metric
+    val merits = metric match {
+      case NLPD =>
+        Map(
+          "Trivial" -> NegativeLogProbabilityDensity2d(0, index, CorrelationMethods.Trivial, observational),
+          "Training Data" -> NegativeLogProbabilityDensity2d(0, index, CorrelationMethods.FromTraining, observational),
+          "Bootstrap" -> NegativeLogProbabilityDensity2d(0, index, CorrelationMethods.Bootstrap, observational),
+          "Jackknife" -> NegativeLogProbabilityDensity2d(0, index, CorrelationMethods.Jackknife, observational)
+        )
+      case StdConfidence =>
+        Map(
+          "Trivial" -> StandardConfidence2d(0, index, CorrelationMethods.Trivial, observational),
+          "Training Data" -> StandardConfidence2d(0, index, CorrelationMethods.FromTraining, observational),
+          "Bootstrap" -> StandardConfidence2d(0, index, CorrelationMethods.Bootstrap, observational),
+          "Jackknife" -> StandardConfidence2d(0, index, CorrelationMethods.Jackknife, observational)
+        )
+    }
+    // There's some unfortunate boiler plate here. plotMeritScan takes a function that goes from a Double parameter value
+    // to an iterator of what is essentially predicted-vs-actuals. This always uses the runTrial method, but for each
+    // case we need to separately pick where the parameter gets plugged in.
+    val pvaBuilder: Double => Iterator[(PredictionResult[Seq[Any]], Seq[Seq[Any]])] = variedParameter match {
```

Comment:
Not the cleanest, but this is just for running your research tests. So I don't see any issue.